implementations/graphql-yoga/src/index.ts

```diff
@@ -1,83 +1,164 @@
-import { buildSubgraphSchema } from '@apollo/subgraph'
-import { createServer } from '@graphql-yoga/node'
+import { buildSubgraphSchema } from "@apollo/subgraph";
+import { createServer, GraphQLYogaError } from "@graphql-yoga/node";
 import { gql } from "graphql-tag";
-import { readFileSync } from 'node:fs'
+import { readFileSync } from "node:fs";
 
-import { Product, Resolvers } from './resolvers-types'
+import { Product, ProductResearch, Resolvers, User } from "./resolvers-types";
 
+const typeDefs = readFileSync("./products.graphql", "utf8");
 
-const typeDefs = readFileSync('./products.graphql', 'utf8')
+const productResearch: ProductResearch[] = [
+  {
+    study: {
+      caseNumber: "1234",
+      description: "Federation Study",
+    },
+  },
+  {
+    study: {
+      caseNumber: "1235",
+      description: "Studio Study",
+    },
+  },
+];
 
-const products: Product[] = [
+const products: Omit<Product, "research">[] = [
   {
     id: "apollo-federation",
     sku: "federation",
     package: "@apollo/federation",
-    variation: {
-      id: "OSS",
-    },
+    variation: { id: "OSS", __typename: "ProductVariation" },
   },
   {
     id: "apollo-studio",
     sku: "studio",
     package: "",
-    variation: {
-      id: "platform",
-    },
+    variation: { id: "platform", __typename: "ProductVariation" },
   },
 ];
 
+const deprecatedProduct = {
+  sku: "apollo-federation-v1",
+  package: "@apollo/federation-v1",
+  reason: "Migrate to Federation V2",
+};
+
+const user: User = {
+  email: "support@apollographql.com",
+  name: "Jane Smith",
+  totalProductsCreated: 1337,
+  yearsOfEmployment: 3,
```

Comment:
based on the https://github.com/apollographql/apollo-federation-subgraph-compatibility/blob/main/CONTRIBUTORS.md#expected-data-sets this should be `10`